cli/cmd/connect.go

```diff
@@ -78,12 +128,32 @@ func resolveInstAddr(cmdCtx *cmdcontext.CmdCtx, cliOpts *config.CliOpts,
 		if len(cmdCtx.Running) > 1 {
 			return newArgs, fmt.Errorf("specify instance name")
 		}
+		if cmdCtx.Connect.Username != "" || cmdCtx.Connect.Password != "" {
+			return newArgs, fmt.Errorf("username and password do not supported" +
+				" with a connection via a control socket")
+		}
 		newArgs[0] = cmdCtx.Running[0].ConsoleSocket
 		return newArgs, nil
-	} else {
-		if isURI(newArgs[0]) {
-			return newArgs, nil
+	} else if isCredentialsURI(newArgs[0]) {
+		if cmdCtx.Connect.Username != "" || cmdCtx.Connect.Password != "" {
+			return newArgs, fmt.Errorf("username and password specified with flags" +
+				" and a URI")
 		}
+		uri, user, pass := parseCredentialsURI(newArgs[0])
```

Comment:
Do we need to update this function comments in accordance with new changes?

Reply:
It's a good idea. I renamed the function and updated the description.